src/openrct2/world/map_generator/HeightMap.hpp

```diff
@@ -24,17 +24,27 @@ namespace OpenRCT2::World::MapGenerator
         std::vector<uint8_t> _height;
 
     public:
-        const uint16_t width;
-        const uint16_t height;
+        uint16_t width{};
+        uint16_t height{};
+        uint8_t density{};
 
-        HeightMap(int32_t tagetWidth, int32_t targetHeight)
-            : _height(tagetWidth * targetHeight)
-            , width(tagetWidth)
+        HeightMap(int32_t targetWidth, int32_t targetHeight)
+            : _height(targetWidth * targetHeight)
+            , width(targetWidth)
             , height(targetHeight)
+            , density(1)
         {
         }
 
-        HeightMap(const HeightMap& heightMap) = default;
+        HeightMap(int32_t baseWidth, int32_t baseHeight, uint8_t density_)
+            : _height((baseWidth * density_) * (baseHeight * density_))
```

Comment:
Is this still correct, considering it also saves the density now?

Reply:
Yes. It's not the best API, but the height map's dimensions are just enlarged to accommodate the intended density. The data still needs to be read at the higher resolution, so this seemed like the best solution, at least until the users are reworked.